plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -357,6 +358,62 @@ func TestGetRoundTripper(t *testing.T) {
 	}
 }
 
+func TestGetRoundTripperToken(t *testing.T) {
+	tokenFilePath := "test/test.txt"
+	wantBearer := "token from file"
+	// Create temp file with token if tokenFilePath is provided
+	dir, file := filepath.Split(tokenFilePath)
+	err := os.MkdirAll(dir, 0o750)
+	require.NoError(t, err)
+	err = os.WriteFile(filepath.Join(dir, file), []byte(wantBearer), 0o660)
+	require.NoError(t, err)
+	defer func() {
+		err = os.RemoveAll(tokenFilePath)
+		require.NoError(t, err)
+	}()
+
+	rt, err := getHTTPRoundTripper(&config.Configuration{
+		ServerURL:      "https://localhost:1234",
```

Comment:
why not call this *after* the test server is created and its ServerURL is known?

Reply:
Yep , since url is not required i have simplified and removed it